src/Microsoft.OData.Client/Materialization/MaterializerEntry.cs

```diff
@@ -314,6 +314,25 @@ public void UpdateEntityDescriptor()
                     }
                 }
 
+                // We need this to be populated as well for entities that may contain this
+                if (this.entry.MediaResource != null)
+                {
+                    if (this.entry.MediaResource.ReadLink != null)
+                    {
+                        this.EntityDescriptor.ReadStreamUri = this.entry.MediaResource.ReadLink;

```

Comment:
Is there a default value (non null) that you are protecting by needing to test the new value for non null first?
Same for the other ifs below this...